qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,380 @@
+                Qiskit :class:`.Optimizer` or a callable implementing the :class:`.Minimizer`
+                protocol. Defaults to :class:`.SLSQP`.
+            gradient: An optional gradient function or operator for the optimizer. initial_point: An
+            optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` then VQE will look to the ansatz for a preferred
+                point and if not will simply compute a random one.
+            callback: A callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the ansatz, the
+                evaluated mean and the evaluated standard deviation.
+        """
+        super().__init__()
+
+        self.estimator = estimator
+        self.ansatz = ansatz
+        self.optimizer = optimizer
+        self.gradient = gradient
+        # this has to go via getters and setters due to the VariationalAlgorithm interface
+        self.initial_point = initial_point
+        self.callback = callback
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
+        """Return the initial point."""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, value: Sequence[float] | None) -> None:
+        """Set the initial point."""
+        self._initial_point = value
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> VQEResult:
+        # set defaults
+        if self.ansatz is None:
+            ansatz = RealAmplitudes(num_qubits=operator.num_qubits)
+        else:
+            ansatz = self.ansatz
+
+        ansatz = self._check_operator_ansatz(operator, ansatz)
+
+        optimizer = SLSQP() if self.optimizer is None else self.optimizer
+
+        initial_point = _validate_initial_point(self.initial_point, ansatz)
+
+        start_time = time()
+
+        evaluate_energy = self._get_evaluate_energy(ansatz, operator)
+
+        if self.gradient is not None:
+            evaluate_gradient = self._get_evalute_gradient(ansatz, operator)
+        else:
+            evaluate_gradient = None
+
+        # perform optimization
+        if callable(optimizer):
+            opt_result = optimizer(fun=evaluate_energy, x0=initial_point, jac=evaluate_gradient)
+        else:
+            opt_result = optimizer.minimize(
+                fun=evaluate_energy, x0=initial_point, jac=evaluate_gradient
+            )
+
+        eval_time = time() - start_time
+
+        result = VQEResult()
+        result.eigenvalue = opt_result.fun + 0j
```

Comment:
Why is this made complex?

Reply:
It's not strictly necessary. The old VQE code made it explicitly complex, so I just left it in. But I can take it out.

Reply:
Interesting.. @Cryoris @woodsp-ibm any insights on why/whether this is needed?

Reply:
It may be from way back when results etc were turned into JSON. We had this value/type in common with Numpy Eigensolver and we did not constrain things there to hermitian but allowed complex results. Opflow did as well. This was probably made complex to be consistent for the convertor etc. I cannot recall exactly. Seems like something we should get rid of now.